xl-v1-multipart.go

```diff
@@ -631,9 +631,11 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 
 	// Rename if an object already exists to temporary location.
 	uniqueID := getUUID()
-	err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
-	if err != nil {
-		return "", toObjectErr(err, bucket, object)
+	if xl.isObject(bucket, object) {
+		err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
```

Comment:
`renameObject()` call in putObjectPart in xl-v1-multipart.go (L390) is probably the reason for the problem. I think `tmpPartPath` is the path to the part (which will give false in the xl.isObject check) and therefore renameObject will return without renaming.